utils/preferences/tests/dtf.rs

```diff
@@ -0,0 +1,85 @@
+use icu_datetime::options::preferences::HourCycle;
+use icu_locid::{unicode_ext_key, Locale};
+use icu_preferences::preferences;
+
+#[derive(Clone, Copy)]
+enum Calendar {
+    Gregory,
+}
+
+impl TryFrom<&icu_locid::extensions::unicode::Value> for Calendar {
+    type Error = ();
+
+    fn try_from(_i: &icu_locid::extensions::unicode::Value) -> Result<Self, Self::Error> {
+        Ok(Calendar::Gregory)
+    }
+}
+
+preferences!(
+    DTFPreferences,
+    ResolvedDTFPreferences,
+    {
+        hour_cycle => Option<HourCycle>, HourCycle, Some(unicode_ext_key!("hc")),
+        calendar => Option<Calendar>, Calendar, Some(unicode_ext_key!("ca"))
+    }
+);
+
+struct DateTimeFormat {
+    prefs: ResolvedDTFPreferences,
+}
+
+impl DateTimeFormat {
+    pub fn new(locale: &Locale, prefs: &DTFPreferences) -> Self {
```

Comment:
Discussion: If we add the preferences bag as a separate argument, we end up with every formatter constructor taking 4 arguments: locale, provider, options, and preferences.

Reply:
I don't think we should. I'm ok with preferences being part of options. This is just an example.